src/renderer/atlas/AtlasEngine.cpp

```diff
@@ -641,12 +633,15 @@ void AtlasEngine::_createResources()
         _r.deviceContext = deviceContext.query<ID3D11DeviceContext1>();
     }
 
-    // > You should not use GetSystemMetrics(SM_REMOTESESSION) to determine if your application is running
-    // > in a remote session in Windows 8 and later or Windows Server 2012 and later if the remote session
-    // > may also be using the RemoteFX vGPU improvements to the Microsoft Remote Display Protocol (RDP).
-    // > In this case, GetSystemMetrics(SM_REMOTESESSION) will identify the remote session as a local session.
-    // This actually sounds great for us. The non-d2dMode of AtlasEngine has more features, but requires a GPU.
-    _r.d2dMode = debugForceD2DMode || GetSystemMetrics(SM_REMOTESESSION);
+    {
+        wil::com_ptr<IDXGIAdapter1> dxgiAdapter;
+        THROW_IF_FAILED(_r.device.query<IDXGIObject>()->GetParent(__uuidof(dxgiAdapter), dxgiAdapter.put_void()));

```

Comment:
nit: i'm surprised `IID_PPV_ARGS` didn't work here.. or would it?

Reply:
It would but most code in this class doesn't use it so I kept the pattern.

Reply:
(The reason I did this is because I wanted more (and more obvious) control over whether `.put()`, `.addressof()`, etc. are used.)